x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Max.java

```diff
@@ -107,7 +109,8 @@ protected TypeResolution resolveType() {
 
     @Override
     public DataType dataType() {
-        return field().dataType();
+        DataType t = field().dataType();
+        return t == TEXT ? KEYWORD : t;
```

Comment:
I have a small suggestion - feel free to leave this as it is - instead of checking for `t == TEXT` we could do `DataType.isString(t) ? KEYWORD : t`. The intention would be a bit clearer that for string arguments, the return type is keyword.
This would work in case we want to add another string type - which we do, we are working on adding `semantic_text`. So I think that even if we leave this as it is for now, we would still need to change this check to use `DataType.isString` when we add support for `semantic_text` to be a string type.

Reply:
I double-checked, and this particular case was indeed not switched to use `noText()`, so I'll fix that directly, and look for more cases I might have missed.